src/main.ts

```diff
@@ -215,4 +219,41 @@ function dependabotJobUrl(id: number): string {
   return url_parts.filter(Boolean).join('/')
 }
 
+function credentialsFromEnv(): Credential[] {
+  const registriesProxyStr = process.env.GITHUB_REGISTRIES_PROXY
+  let credentialsStr: string
+  if (registriesProxyStr !== undefined) {
+    credentialsStr = Buffer.from(registriesProxyStr, 'base64').toString()
+  } else {
+    return []
+  }
+
+  let parsed: Credential[]
+
+  try {
+    parsed = JSON.parse(credentialsStr) as Credential[]
+  } catch {
+    // Don't log the error as it may contain sensitive information
+    parsed = []
+    botSay('Failed to parse GITHUB_REGISTRIES_PROXY environment variable')
+  }
+
+  for (const e of parsed) {
+    // Mask credentials to reduce chance of accidental leakage in logs.
+    if (e.password !== undefined) {
+      core.setSecret(e.password)
+    }
+    if (e.token !== undefined) {
+      core.setSecret(e.token)
+    }
+    if (e['auth-key'] !== undefined) {
+      core.setSecret(e['auth-key'])
+    }
+
+    // TODO: Filter down to only credentials relevant to this job.
```

Comment:
At the rate we're adding ecosystems, some of which overlap (e.g. dotnet_sdk and nuget) I think we should inject all secrets to all job types.